mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -4841,17 +4853,22 @@ struct TransformRewritePattern : public OpRewritePattern<miopen::TransformOp> {
             break;
 
         auto coordTransformAttrs = user->getAttr("coord_transforms");
-        if (!coordTransformAttrs)
-          user->setAttr(
-              "coord_transforms",
-              b.getArrayAttr({b.getDictionaryAttr(
-                  {b.getNamedAttr("operand",
-                                  b.getI32IntegerAttr(userOperandIndex)),
-                   b.getNamedAttr(
-                       "transforms",
-                       b.getAffineMapArrayAttr(outputType.getAffineMaps())),
-                   b.getNamedAttr("domain", b.getArrayAttr(shapeAttrVec))})}));
-        else {
+        if (!coordTransformAttrs) {
+          llvm::SmallVector<NamedAttribute, 4> arrayAttr{
+              b.getNamedAttr("operand", b.getI32IntegerAttr(userOperandIndex)),
+              b.getNamedAttr("transforms", b.getAffineMapArrayAttr(
+                                               outputType.getAffineMaps()[0])),
+              b.getNamedAttr("domain", b.getArrayAttr(shapeAttrVec))};
+          if (outputType.getAffineMaps().size() > 1)
+            arrayAttr.push_back(b.getNamedAttr(
+                "limits",
+                b.getAffineMapArrayAttr(outputType.getAffineMaps()[1])));
```

Comment:
what would happen in case the dimension numbers are different between `miopen.transform`?

for example: 2D -> 4D , 4D -> 2D?

could you give an example here?